### PR TITLE
More serialization annotations

### PR DESCRIPTION
**Pull request by stephentoub**
Mostly just adding [Serializable] around.  In a few places there's a bit more implementation with it.  And in the permissions library, I took the opportunity to let VS do some of the automatic cleanup it wanted to.

Contributes to https://github.com/dotnet/corefx/issues/12725
cc: @danmosemsft

### CODE REVIEW

File: src/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs [hunk 1]
```diff
@@ -10,11 +10,13 @@ namespace System.Xml
     /// <devdoc>
     ///    <para>[To be supplied.]</para>
     /// </devdoc>
+    [Serializable]
     public class XmlQualifiedName
     {
         private string _name;
         private string _ns;
 
+        [NonSerialized]
```
Comment (danmosemsft) [inline src/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs]: >NonSerialized [](start = 9, length = 13)

would your tool flag any places missing this (or that have it and shouldn't)
I didn't check.

**Comment by danmosemsft** [comment]
:shipit:

File: src/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs]: It won't.  I'd need to modify it to handle that, which I expect would be mostly doable but not quite as easy, as we've renamed many of the fields.  We can explore doing so though.

**Comment by stephentoub** [comment]
Thanks for reviewing, @danmosemsft.

**Pull request merged**